dot.cpp

```diff
@@ -66,33 +78,64 @@ Dot::Dot(const Dot &d, bool exact_copy /* = true*/):
 }
 
 void Dot::tick(){
-  if (food_in_sight_) {
-    if(food_in_sight_dist_ < speed_){
-      position_ = food_in_sight_pos_;
-      energy_ -= food_in_sight_dist_ * speed_;
+  // First priority is to flee from hazards
+  if (hazard_in_sight_){
+    if (distance(position_, hazard_in_sight_pos_, testground_size_) > sight_ * 2){
+      hazard_in_sight_ = false;
     } else {
-      direction_ = direction(position_, food_in_sight_pos_, testground_size_);
+      direction_ = (direction(position_, food_in_sight_pos_, testground_size_) + 180) % 360;
       position_ = move(position_, direction_, speed_, testground_size_);
       energy_ -= pow(speed_, 2);
     }
-  } else {
-    direction_ += ((rand() % 21) - 10) % 360;
-    position_ = move(position_, direction_, speed_, testground_size_);
-    energy_ -= pow(speed_, 2);
   }
-
+  // This can't be just an else-block since hazard_in_sight_ may change in the first block to false
+  if(!hazard_in_sight_){
+    // Second priority is to hunt preys
+    if(prey_in_sight_){
+      if(prey_in_sight_dist_ < speed_){
+        position_ = prey_in_sight_pos_;
+        energy_ -= prey_in_sight_dist_ * speed_;
+        // TODO: Trigger deletion of Prey
+      } else {
+        direction_ = direction(position_, prey_in_sight_pos_, testground_size_);
+        position_ = move(position_, direction_, speed_, testground_size_);
+        energy_ -= pow(speed_, 2);
+      }
```

Comment:
prey_in_sight_pos_ might have changed in between since the prey might have moved before the hunter. Unfortunately I don't think that this can be easily fixed. The Dot can't access the other Dot with the current implementation. To fix this, the hunter would need to store the prey as a reference, which might cause trouble, when Dots are deleted and also isn't very realistic. If we give the hunter a reference, the Dot can theoretically access all the data of the prey at any time.